x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/action/enrollment/TransportKibanaEnrollmentActionTests.java

```diff
@@ -0,0 +1,169 @@
+            nodeInfos.add(new NodeInfo(Version.CURRENT,
+                null,
+                n,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                new HttpInfo(new BoundTransportAddress(new TransportAddress[] { n.getAddress() }, n.getAddress()), randomLong()),
+                null,
+                null,
+                null,
+                null));
+        }
+        doAnswer(invocation -> {
+            NodesInfoRequest nodesInfoRequest = (NodesInfoRequest) invocation.getArguments()[1];
+            nodesInfoRequests.add(nodesInfoRequest);
+            ActionListener<NodesInfoResponse> listener = (ActionListener) invocation.getArguments()[2];
+            listener.onResponse(new NodesInfoResponse(new ClusterName("cluster"), nodeInfos, List.of()));
+            return null;
+        }).when(client).execute(eq(NodesInfoAction.INSTANCE), any(), any());
+
+        doAnswer(invocation -> {
+            ChangePasswordRequest changePasswordRequest = (ChangePasswordRequest) invocation.getArguments()[1];
+            changePasswordRequests.add(changePasswordRequest);
+            ActionListener<ActionResponse.Empty> listener = (ActionListener) invocation.getArguments()[2];
+            listener.onResponse(ActionResponse.Empty.INSTANCE);
+            return null;
+        }).when(client).execute(eq(ChangePasswordAction.INSTANCE), any(), any());
+
+        final TransportService transportService = new TransportService(Settings.EMPTY,
+            mock(Transport.class),
+            threadPool,
+            TransportService.NOOP_TRANSPORT_INTERCEPTOR,
+            x -> null,
+            null,
+            Collections.emptySet());
+        action = new TransportKibanaEnrollmentAction(transportService, client, settings, sslService, env, mock(ActionFilters.class));
+    }
+
+    public void testKibanaEnrollment() {
+        final KibanaEnrollmentRequest request = new KibanaEnrollmentRequest();
+        final PlainActionFuture<KibanaEnrollmentResponse> future = new PlainActionFuture<>();
+        action.doExecute(mock(Task.class), request, future);
+        final KibanaEnrollmentResponse response = future.actionGet();
+        assertThat(response.getHttpCa(), startsWith("MIIDSjCCAjKgAwIBAgIVALCgZXvbceUrjJaQMheDCX0kXnRJMA0GCSqGSIb3DQEBCwUAMDQxMjAw" +
+            "BgNVBAMTKUVsYXN0aWMgQ2VydGlmaWNhdGUgVG9vbCBBdXRvZ2VuZXJhdGVkIENBMB4XDTIxMDQyODEyNTY0MVoXDTI0MDQyNzEyNTY0MVowNDEyMDAGA1UEA" +
+            "xMpRWxhc3RpYyBDZXJ0aWZpY2F0ZSBUb29sIEF1dG9nZW5lcmF0ZWQgQ0EwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQCCJbOU4JvxDD_F"));
+        assertThat(response.getNodesAddresses().size(), equalTo(numberOfNodes));
+        assertThat(changePasswordRequests.size(), equalTo(1));
+        assertThat(nodesInfoRequests.size(), equalTo(1));
+
+        assertWarnings("[keystore.password] setting was deprecated in Elasticsearch and will be removed in a future release! " +
```

Comment:
Yes we needed to.  This is not part of this test, as in we're not testing something with this assertion but without assertWarnings there , the test fails. I'll replace the setup to use secure settings so that this is not needed any more